building/build-debs/homeworld-admin-tools/resources/postinstall.sh

```diff
@@ -40,17 +40,34 @@ cp /keyclient-*.yaml /target/etc/homeworld/config/
 cp /keyclient-base.yaml /target/etc/homeworld/config/keyclient.yaml
 cp /sshd_config.new /target/etc/ssh/sshd_config
 
-cat >/tmp/token.template <<EOF
-Template: homeworld/asktoken
+generate_token_template () {
+    cat >/tmp/$1.template <<EOF
+Template: homeworld/ask$1
 Type: string
-Description: Enter the bootstrap token for this server.
+Description: $2
 
 Template: homeworld/title
 Type: text
 Description: Configuring keysystem...
 EOF
 
-debconf-loadtemplate homeworld /tmp/token.template
+}
+
+generate_token_template "token" "Enter the bootstrap token for this server."
+generate_token_template "tokeninvalid" "Invalid token! Please check the token for typos."
+debconf-loadtemplate homeworld /tmp/token.template /tmp/tokeninvalid.template
+
+validate_token () {
+    TOKEN="${1%??}"
+    TOKEN_PROVIDED_HASH=`echo -n $1 | tail -c 2`
+    TOKEN_ACTUAL_HASH=`echo -n "$TOKEN" | /target/usr/bin/openssl dgst -sha256 -binary | /target/usr/bin/base64 | cut -c -2`
+
+    if [ "$TOKEN_PROVIDED_HASH" != "$TOKEN_ACTUAL_HASH" ]; then
+        return 0
+    fi
+
+    return 1
+}   
```

Comment:
The name "validate_token" implies that it returns true when valid and false when invalid, whereas this does the opposite. Perhaps a different name would make this better?